src/openrct2/core/FileStream.hpp

```diff
@@ -81,6 +82,16 @@ namespace OpenRCT2
                     throw;
             }
 
+            // Make sure the directory exists before writing to a file inside it
+            if (_canWrite)
+            {
+                std::string directory = Path::GetDirectory(path);
+                if (!Path::DirectoryExists(directory))
+                {
+                    Path::CreateDirectory(directory);
```

Comment:
Shouldn't we check the return and throw in case this fails?

Reply:
`Path::CreateDirectory` is void and doesn't throw from the looks of it. It itself does call a function returning bool, however. It's probably a good idea to validate the return value there and return false or throw on failure.

Reply:
I don't mind if this is left for another PR, as it will take changing multiple places.